guppylang/tracing/function.py

```diff
@@ -162,10 +161,10 @@ def trace_call(func: CompiledCallableDef, *args: Any) -> Any:
 
     # Check call
     arg_exprs: list[ast.expr] = [
-        with_loc(func.defined_at, with_type(var.ty, PlaceNode(var))) for var in arg_vars
+        with_loc(state.node, with_type(var.ty, PlaceNode(var))) for var in arg_vars
```

Comment:
Instead, switch to the node we have stored in the tracing state. It doesn't really matter since we're not using the span-renderer for comptime errors anyways